### ISSUE TITLE
Test fixed Arch Linux VDT feed

### ISSUE THREAD

**Issue by Deblintrake09** [wazuh/wazuh-qa#3504]
| Target version | Related issue | Related PR |
|--------------------|--------------------|-----------------|
| 4.3.10 |  https://github.com/wazuh/wazuh/issues/15180                      | [15180-arch-linux-url   ](https://github.com/wazuh/wazuh/pull/15219)                |

## Description
The Vulnerability Detector feed for the Arch Linux OS was outdated and has been changed. This issue aims to test that VDT downloads the feeds and works correctly for that OS family.

## Proposed checks
- [x] Launch ITs
- [x] Test VDT generates alerts for Arch Linux agent

## Steps to reproduce

- Setup manager and Arch Linux agent
- Enable VDT
- Wait for Scan and check alerts have been generated

## Expected results
- The Arch Linux feed is downloaded correctly.
- The VDT scan is executed correctly and generates alerts.

**Comment by Deblintrake09** [comment]
- Continue researching for boxes
- No available Archlinux box in AWS
- vagrant boxes all fail ssh connection
```
vagrant up Arch
Bringing machine 'Arch' up with 'virtualbox' provider...
==> Arch: Box 'terrywang/archlinux' could not be found. Attempting to find and install...
    Arch: Box Provider: virtualbox
    Arch: Box Version: >= 0
==> Arch: Loading metadata for box 'terrywang/archlinux'
    Arch: URL: https://vagrantcloud.com/terrywang/archlinux
==> Arch: Adding box 'terrywang/archlinux' (v3.22.0817) for provider: virtualbox
    Arch: Downloading: https://vagrantcloud.com/terrywang/boxes/archlinux/versions/3.22.0817/providers/virtualbox.box
    Arch: Download redirected to host: vagrantcloud-files-production.s3-accelerate.amazonaws.com
    Arch: Calculating and comparing box checksum...
==> Arch: Successfully added box 'terrywang/archlinux' (v3.22.0817) for 'virtualbox'!
==> Arch: Importing base box 'terrywang/archlinux'...
==> Arch: Matching MAC address for NAT networking...
==> Arch: Checking if box 'terrywang/archlinux' version '3.22.0817' is up to date...
==> Arch: Setting the name of the VM: Vagrant_Arch_1666814397570_93974
==> Arch: Clearing any previously set network interfaces...
==> Arch: Preparing network interfaces based on configuration...
    Arch: Adapter 1: nat
    Arch: Adapter 2: hostonly
==> Arch: Forwarding ports...
    Arch: 22 (guest) => 5555 (host) (adapter 1)
==> Arch: Running 'pre-boot' VM customizations...
==> Arch: Booting VM...
==> Arch: Waiting for machine to boot. This may take a few minutes...
    Arch: SSH address: 127.0.0.1:5555
    Arch: SSH username: vagrant
    Arch: SSH auth method: private key
    Arch: Warning: Authentication failure. Retrying...
    Arch: Warning: Authentication failure. Retrying...
    Arch: Warning: Authentication failure. Retrying...
    Arch: Warning: Authentication failure. Retrying...



$ ping 10.0.10.254
PING 10.0.10.254 (10.0.10.254) 56(84) bytes of data.
From 10.0.10.1 icmp_seq=1 Destination Host Unreachable
From 10.0.10.1 icmp_seq=2 Destination Host Unreachable
```

**Comment by Deblintrake09** [comment]
# Task Results

| Path | Results | Notes|
|---|---|---|
| test_vulnerability_detector/ | [:large_blue_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/33250/) [:large_blue_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/33251/) [:large_blue_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/33252/) |

**Comment by Deblintrake09** [comment]
# Testing results

## Work on Fixed branch

<details><summary> Launch ITs :green_circle: </summary
| Path | Results | Notes|
|---|---|---|
| test_vulnerability_detector/ | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/33298/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/33300/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/33299/) |
</details>

<details><summary> Scan Archlinux agent :green_circle: </summary

1. Install manager
2. Install archlinux agent
```
# /var/ossec/bin/agent_control -i 001

Wazuh agent_control. Agent information:
   Agent ID:   001
   Agent Name: d4b9460da662
   IP address: any/any
   Status:     Active

   Operating system:    Linux |d4b9460da662 |5.14.0-1054-oem |#61-Ubuntu SMP Fri Oct 14 13:05:50 UTC 2022 |x86_64
   Client version:      Wazuh v4.3.9
   Configuration hash:  ab73af41699f13fdd81903b5f23d8d00
   Shared file hash:    4a8724b20dee0124ff9656783c490c4e
   Last keep alive:     1666885744

   Syscheck last started at:  Thu Oct 27 14:59:30 2022
   Syscheck last ended at:    Thu Oct 27 14:59:30 2022
```
```
[root@d4b9460da662 wazuh]# cat /etc/os-release 
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
VERSION_ID=TEMPLATE_VERSION_ID
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://bugs.archlinux.org/"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo

```
3. Wait for scan
```
2022/10/27 15:01:36 wazuh-modulesd:vulnerability-detector: INFO: (5400): Starting 'Arch Linux' database update.
2022/10/27 15:01:40 wazuh-modulesd:vulnerability-detector: INFO: (5430): The update of the 'Arch Linux' feed finished successfully.
2022/10/27 15:01:40 wazuh-modulesd:vulnerability-detector: INFO: (5400): Starting 'National Vulnerability Database' database update.
2022/10/27 15:01:56 wazuh-modulesd[5144] debug_op.c:70 at _log(): DEBUG: Logging module auto-initialized
.
.
.
2022/10/27 15:11:30 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:1967 at wm_vuldet_linux_rm_nvd_not_dependencies_met_packages(): DEBUG: (5463): Package 'linux_kernel' not vulnerable to 'CVE-2013-0796' since it don't meet its 'sibling' dependency on package with ID '30907'
2022/10/27 15:11:30 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:1967 at wm_vuldet_linux_rm_nvd_not_dependencies_met_packages(): DEBUG: (5463): Package 'linux_kernel' not vulnerable to 'CVE-2013-0796' since it don't meet its 'sibling' dependency on package with ID '30910'
2022/10/27 15:11:30 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:1967 at wm_vuldet_linux_rm_nvd_not_dependencies_met_packages(): DEBUG: (5463): Package 'linux_kernel' not vulnerable to 'CVE-2013-0796' since it don't meet its 'sibling' dependency on package with ID '30913'
2022/10/27 15:11:30 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:1967 at wm_vuldet_linux_rm_nvd_not_dependencies_met_packages(): DEBUG: (5463): Package 'linux_kernel' not vulnerable to 'CVE-2013-0796' since it don't meet its 'sibling' dependency on package with ID '30916'
2022/10/27 15:11:30 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:1967 at wm_vuldet_linux_rm_nvd_not_dependencies_met_packages(): DEBUG: (5463): Package 'linux_kernel' not vulnerable to 'CVE-2013-0796' since it don't meet its 'sibling' dependency on package with ID '30919'
2022/10/27 15:11:30 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:2127 at wm_vuldet_linux_rm_false_positives(): DEBUG: (5465): A total of '2450' potential vulnerabilities have been discarded for agent '001'
2022/10/27 15:11:30 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:2128 at wm_vuldet_linux_rm_false_positives(): DEBUG: (5470): It took '0' seconds to 'filter' vulnerabilities in agent '001'
2022/10/27 15:11:30 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:1421 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5466): Sending vulnerabilities report for agent '001'
2022/10/27 15:11:31 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:1531 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5468): The 'perl' package (5.36.0-1) from agent '001' is vulnerable to 'CVE-2021-36770'. Condition: 'Package greater than or equal 5.34.0-3'
2022/10/27 15:11:31 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:1531 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5468): The 'openssl' package (1.1.1.q-1) from agent '001' is vulnerable to 'CVE-2022-2068'. Condition: 'Package greater than or equal 1.1.1.o-1'
2022/10/27 15:11:31 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:1531 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5468): The 'wget' package (1.21.3-1) from agent '001' is vulnerable to 'CVE-2021-31879'. Condition: 'Package greater than or equal 1.21.3-1'
2022/10/27 15:11:31 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:1531 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5468): The 'perl' package (5.36.0-1) from agent '001' is vulnerable to 'CVE-2020-16156'. Condition: 'Package greater than or equal 5.34.0-3'
2022/10/27 15:11:31 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:1559 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5482): A total of '1' vulnerabilities have been reported for agent '001' thanks to the 'NVD' feed.
2022/10/27 15:11:31 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:1560 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5482): A total of '4' vulnerabilities have been reported for agent '001' thanks to the 'vendor' feed.
2022/10/27 15:11:31 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:1562 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5469): A total of '4' vulnerabilities have been reported for agent '001'
2022/10/27 15:11:31 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:1563 at wm_vuldet_process_agent_vulnerabilities(): DEBUG: (5470): It took '1' seconds to 'report' vulnerabilities in agent '001'
2022/10/27 15:11:31 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:2603 at wm_vuldet_check_agent_vulnerabilities(): INFO: (5471): Finished vulnerability assessment for agent '001'
2022/10/27 15:11:31 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:2604 at wm_vuldet_check_agent_vulnerabilities(): DEBUG: (5470): It took '2' seconds to 'scan' vulnerabilities in agent '001'
2022/10/27 15:11:31 wazuh-modulesd:vulnerability-detector[5520] wm_vuln_detector.c:7770 at wm_vuldet_run_scan(): INFO: (5472): Vulnerability scan finished.

```
4. Check for alerts
```
{"timestamp":"2022-10-27T15:11:31.861+0000","rule":{"level":13,"description":"CVE-2022-2068 affects openssl","id":"23506","firedtimes":1,"mail":true,"groups":["vulnerability-detector"],"gdpr":["IV_35.7.d"],"pci_dss":["11.2.1","11.2.3"],"tsc":["CC7.1","CC7.2"]},"agent":{"id":"001","name":"d4b9460da662","ip":"172.17.0.2"},"manager":{"name":"c3"},"id":"1666883491.572167","decoder":{"name":"json"},"data":{"vulnerability":{"package":{"name":"openssl","version":"1.1.1.q-1","architecture":"x86_64","condition":"Package greater than or equal 1.1.1.o-1"},"cvss":{"cvss2":{"vector":{"attack_vector":"network","access_complexity":"low","authentication":"none","confidentiality_impact":"complete","integrity_impact":"complete","availability":"complete"},"base_score":"10"},"cvss3":{"vector":{"attack_vector":"network","access_complexity":"low","privileges_required":"none","user_interaction":"none","scope":"unchanged","confidentiality_impact":"high","integrity_impact":"high","availability":"high"},"base_score":"9.800000"}},"cve":"CVE-2022-2068","title":"CVE-2022-2068 affects openssl","rationale":"In addition to the c_rehash shell command injection identified in CVE-2022-1292, further circumstances where the c_rehash script does not properly sanitise shell metacharacters to prevent command injection were found by code review. When the CVE-2022-1292 was fixed it was not discovered that there are other places in the script where the file names of certificates being hashed were possibly passed to a command executed through the shell. This script is distributed by some operating systems in a manner where it is automatically executed. On such operating systems, an attacker could execute arbitrary commands with the privileges of the script. Use of the c_rehash script is considered obsolete and should be replaced by the OpenSSL rehash command line tool. Fixed in OpenSSL 3.0.4 (Affected 3.0.0,3.0.1,3.0.2,3.0.3). Fixed in OpenSSL 1.1.1p (Affected 1.1.1-1.1.1o). Fixed in OpenSSL 1.0.2zf (Affected 1.0.2-1.0.2ze).","severity":"Critical","published":"2022-06-21","updated":"2022-07-23","cwe_reference":"CWE-78","status":"Active","type":"PACKAGE","references":["https://git.openssl.org/gitweb/?p=openssl.git;a=commitdiff;h=9639817dac8bbbaa64d09efad7464ccc405527c7","https://www.openssl.org/news/secadv/20220621.txt","https://git.openssl.org/gitweb/?p=openssl.git;a=commitdiff;h=7a9c027159fe9e1bbc2cd38a8a2914bff0d5abd9","https://git.openssl.org/gitweb/?p=openssl.git;a=commitdiff;h=2c9c35870601b4a44d86ddbf512b38df38285cfa","https://www.debian.org/security/2022/dsa-5169","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6WZZBKUHQFGSKGNXXKICSRPL7AMVW5M5/","https://security.netapp.com/advisory/ntap-20220707-0008/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/VCMNWKERPBKOEBNL7CLTTX3ZZCZLH7XA/","https://nvd.nist.gov/vuln/detail/CVE-2022-2068","https://security.archlinux.org/AVG-2765"],"assigner":"openssl-security@openssl.org","cve_version":"4.0"}},"location":"vulnerability-detector"}
{"timestamp":"2022-10-27T15:11:31.912+0000","rule":{"level":7,"description":"CVE-2021-31879 affects wget","id":"23504","firedtimes":1,"mail":false,"groups":["vulnerability-detector"],"gdpr":["IV_35.7.d"],"pci_dss":["11.2.1","11.2.3"],"tsc":["CC7.1","CC7.2"]},"agent":{"id":"001","name":"d4b9460da662","ip":"172.17.0.2"},"manager":{"name":"c3"},"id":"1666883491.578284","decoder":{"name":"json"},"data":{"vulnerability":{"package":{"name":"wget","version":"1.21.3-1","architecture":"x86_64","condition":"Package greater than or equal 1.21.3-1"},"cvss":{"cvss2":{"vector":{"attack_vector":"network","access_complexity":"medium","authentication":"none","confidentiality_impact":"partial","integrity_impact":"partial","availability":"none"},"base_score":"5.800000"},"cvss3":{"vector":{"attack_vector":"network","access_complexity":"low","privileges_required":"none","user_interaction":"required","scope":"changed","confidentiality_impact":"low","integrity_impact":"low","availability":"none"},"base_score":"6.100000"}},"cve":"CVE-2021-31879","title":"CVE-2021-31879 affects wget","rationale":"GNU Wget through 1.21.1 does not omit the Authorization header upon a redirect to a different origin, a related issue to CVE-2018-1000007.","severity":"Medium","published":"2021-04-29","updated":"2022-05-13","cwe_reference":"CWE-601","status":"Active","type":"PACKAGE","references":["https://mail.gnu.org/archive/html/bug-wget/2021-02/msg00002.html","https://security.netapp.com/advisory/ntap-20210618-0002/","https://nvd.nist.gov/vuln/detail/CVE-2021-31879","https://security.archlinux.org/AVG-1892"],"assigner":"cve@mitre.org","cve_version":"4.0"}},"location":"vulnerability-detector"}
{"timestamp":"2022-10-27T15:11:31.953+0000","rule":{"level":10,"description":"CVE-2020-16156 affects perl","id":"23505","firedtimes":2,"mail":false,"groups":["vulnerability-detector"],"gdpr":["IV_35.7.d"],"pci_dss":["11.2.1","11.2.3"],"tsc":["CC7.1","CC7.2"]},"agent":{"id":"001","name":"d4b9460da662","ip":"172.17.0.2"},"manager":{"name":"c3"},"id":"1666883491.581565","decoder":{"name":"json"},"data":{"vulnerability":{"package":{"name":"perl","version":"5.36.0-1","architecture":"x86_64","condition":"Package greater than or equal 5.34.0-3"},"cvss":{"cvss2":{"vector":{"attack_vector":"network","access_complexity":"medium","authentication":"none","confidentiality_impact":"partial","integrity_impact":"partial","availability":"partial"},"base_score":"6.800000"},"cvss3":{"vector":{"attack_vector":"local","access_complexity":"low","privileges_required":"none","user_interaction":"required","scope":"unchanged","confidentiality_impact":"high","integrity_impact":"high","availability":"high"},"base_score":"7.800000"}},"cve":"CVE-2020-16156","title":"CVE-2020-16156 affects perl","rationale":"CPAN 2.28 allows Signature Verification Bypass.","severity":"High","published":"2021-12-13","updated":"2022-04-01","cwe_reference":"CWE-347","status":"Active","type":"PACKAGE","references":["https://metacpan.org/pod/distribution/CPAN/scripts/cpan","https://blog.hackeriet.no/cpan-signature-verification-vulnerabilities/","http://blogs.perl.org/users/neilb/2021/11/addressing-cpan-vulnerabilities-related-to-checksums.html","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/SZ32AJIV4RHJMLWLU5QULGKMMIHYOMDC/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/SD6RYOJII7HRJ6WVORFNVTYNOFY5JDXN/","https://nvd.nist.gov/vuln/detail/CVE-2020-16156","https://security.archlinux.org/AVG-2630"],"assigner":"cve@mitre.org","cve_version":"4.0"}},"location":"vulnerability-detector"}
```
</details>

**Comment by jmv74211** [comment]
## Closing conclusion 👍🏼  

Everything seems to be working properly.
